hal/src/lib.rs

```diff
@@ -72,6 +72,8 @@ pub mod pwm;
 pub mod pwm;
 
 pub mod delay;
+#[cfg(feature="unproven")]
+pub mod dma;
```

Comment:
Gated this under `unproven` for now, but perhaps it could use its own cargo feature, e.g. `dma`